src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -1,113 +1,213 @@
-  noResultsFoundText: 'No results found',
-  loadingText: 'Loading',
+  suggestionsHeaderText: "Suggested People",
+  mostRecentlyUsedHeaderText: "Suggested Contacts",
+  noResultsFoundText: "No results found",
+  loadingText: "Loading",
   showRemoveButtons: true,
-  suggestionsAvailableAlertText: 'People Picker Suggestions available',
-  suggestionsContainerAriaLabel: 'Suggested contacts',
+  suggestionsAvailableAlertText: "People Picker Suggestions available",
+  suggestionsContainerAriaLabel: "Suggested contacts",
 };
 
-export const PeoplePicker: React.FunctionComponent<any> = (props) => {
-  const [currentSelectedItems, setCurrentSelectedItems] = React.useState<IPersonaProps[]>([]);
-  const [peopleList] = React.useState<IPersonaProps[]>(people);
+export interface SPPersona extends IPersonaProps {
+  AccountName?: string;
+  Department?: string;
+  Email?: string;
+  SPUserId?: number;
+}
+
+interface IPeoplePickerProps {
+  /** Required - The text used to label this people picker for screenreaders */
+  ariaLabel: string;
+  /** Optional - The people to pre-populate the People Picker with */
+  defaultValue?: SPPersona[];
+  readOnly?: boolean;
+  required?: boolean;
+  /** Optional - Limit the People Picker to only allow selection of specific number -- Defaults to 1 */
+  itemLimit?: number;
+  updatePeople: (p: SPPersona[]) => void;
+}
+
+export const PeoplePicker: FunctionComponent<IPeoplePickerProps> = (props) => {
+  const [currentSelectedItems, setCurrentSelectedItems] = useState<
+    IPersonaProps[]
+  >([]);
+  const [peopleList] = useState<IPersonaProps[]>(people);
 
-  const picker = React.useRef(null);
+  useEffect(() => {
+    let personas: SPPersona[] = [];
+    if (props.defaultValue) {
+      personas = [...props.defaultValue];
+    }
+    setCurrentSelectedItems(personas);
+  }, [props.defaultValue]);
 
-  const onFilterChanged = (
+  const picker = useRef<IBasePicker<IPersonaProps>>(null);
+
+  const onFilterChanged = async (
     filterText: string,
     currentPersonas?: IPersonaProps[],
     limitResults?: number,
     selectedPersonas?: IPersonaProps[] | undefined
-  ): IPersonaProps[] | Promise<IPersonaProps[]>  => {
+    //): IPersonaProps[] | Promise<IPersonaProps[]>  => {
+  ): Promise<IPersonaProps[]> => {
     if (filterText) {
-      let filteredPersonas: IPersonaProps[] = filterPersonasByText(filterText);
-      if(currentPersonas)
-      {
+      let filteredPersonas: IPersonaProps[]; //| Promise<IPersonaProps[]>;
+      if (process.env.NODE_ENV === "development") {
+        filteredPersonas = await filterPersonasByText(filterText);
+      } else {
+        const results =
+          await spWebContext.profiles.clientPeoplePickerSearchUser({
+            AllowEmailAddresses: false,
+            AllowMultipleEntities: false,
+            MaximumEntitySuggestions: limitResults ? limitResults : 25,
+            QueryString: filterText,
+            PrincipalSource: 15,
+            PrincipalType: 1,
```

Comment:
Should use the enums instead of their values here
```
PrincipalSource: PrincipalSource.All,
PrincipalType: PrincipalType.User,
```

Reply:
Another good catch -- lodged #17